cluster/src/main/java/io/scalecube/cluster/membership/MembershipProtocol.java

```diff
@@ -238,13 +236,13 @@ public void stop() {
    * ============== Action Methods ================== *
    * ================================================ */
 
-  private ListenableFuture<Void> doInitialSync() {
+  private CompletableFuture<Void> doInitialSync() {
     LOGGER.debug("Making initial Sync to all seed members: {}", seedMembers);
     if (seedMembers.isEmpty()) {
-      return Futures.immediateFuture(null);
+      return CompletableFuture.completedFuture(null);
```

Comment:
hm, actually i see a bug here (not related to futures). We need to call `schedulePeriodicSync();` before returning future.